src/main/java/net/minestom/server/permission/PermissionHandler.java

```diff
@@ -1,138 +1,17 @@
 package net.minestom.server.permission;
 
 import org.jetbrains.annotations.NotNull;
-import org.jetbrains.annotations.Nullable;
-import org.jglrxavpok.hephaistos.nbt.NBTCompound;
-import org.jglrxavpok.hephaistos.parser.SNBTParser;
-
-import java.util.Set;
-import java.util.regex.Pattern;
 
 /**
  * Represents an object which can have permissions.
- * <p>
- * Permissions are in-memory only by default.
- * You have however the capacity to store them persistently as the {@link Permission} object
- * is serializer-friendly, {@link Permission#getPermissionName()} being a {@link String}
- * and {@link Permission#getNBTData()} serializable into a string using {@link NBTCompound#toSNBT()}
- * and deserialized back with {@link SNBTParser#parse()}.
  */
+@FunctionalInterface
 public interface PermissionHandler {
-
-    /**
-     * Returns all permissions associated to this handler.
-     * The returned collection should be modified only by subclasses.
-     *
-     * @return the permissions of this handler.
-     */
-    @NotNull
-    Set<Permission> getAllPermissions();
-
-    /**
-     * Adds a {@link Permission} to this handler.
-     *
-     * @param permission the permission to add
-     */
-    default void addPermission(@NotNull Permission permission) {
-        getAllPermissions().add(permission);
-    }
-
-    /**
-     * Removes a {@link Permission} from this handler.
-     *
-     * @param permission the permission to remove
-     */
-    default void removePermission(@NotNull Permission permission) {
-        getAllPermissions().remove(permission);
-    }
-
-    /**
-     * Removes a {@link Permission} based on its string identifier.
-     *
-     * @param permissionName the permission name
-     */
-    default void removePermission(@NotNull String permissionName) {
```

Comment:
For a none-breaking version, shouldn't this be readded (to this interface) and implemented in DefaultPermissionHandler, so existing code can expect the same results.